### ISSUE TITLE
[Build Failed]glib

### ISSUE THREAD

**Issue by YunchengLiu** [microsoft/vcpkg#27235]
**Host Environment**

- OS:   Windows 11 21H2 22000.1098
- Compiler: MSVC 17.3.6

**To Reproduce**

`.\vcpkg.exe install glib:x64-windows`

**Failure logs**

Computing installation plan...
The following packages will be built and installed:
    glib[core]:x64-windows -> 2.74.0
Detecting compiler hash for triplet x64-windows...
-- Automatically setting HTTP(S)_PROXY environment variables to "127.0.0.1:7890".
Restored 0 package(s) from C:\Users\YunchengLiu\AppData\Local\vcpkg\archives in 123.2 us. Use --debug to see more details.
Installing 1/1 glib:x64-windows...
Building glib[core]:x64-windows...
-- Using cached glib-2.74.0.tar.xz.
-- Cleaning sources at C:/_DevEnv/vcpkg/buildtrees/glib/src/glib-2-22354247da.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/_DevEnv/vcpkg/downloads/glib-2.74.0.tar.xz
-- Applying patch use-libiconv-on-windows.patch
-- Applying patch libintl.patch
-- Using source at C:/_DevEnv/vcpkg/buildtrees/glib/src/glib-2-22354247da.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-windows
-- Configuring x64-windows-dbg
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at C:/_DevEnv/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x64-windows-dbg done
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Package x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" install -v
    Working Directory: C:/_DevEnv/vcpkg/buildtrees/glib/x64-windows-dbg
    Error code: 1
    See logs for more information:
      C:\_DevEnv\vcpkg\buildtrees\glib\package-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/glib/portfile.cmake:47 (vcpkg_install_meson)
  scripts/ports.cmake:147 (include)

**Output Logs**
[package-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/9788399/package-x64-windows-dbg-out.log)

**Comment by Sammot** [comment]
Same problem, seems to be a recent issue since it used to work for me on my other PC.

**Comment by FrankXie05** [comment]
@YunchengLiu Thanks for posting this issue, 
````
Traceback (most recent call last):

  File "C:\_DevEnv\vcpkg\buildtrees\glib\x64-windows-dbg\gobject\glib-mkenums", line 486, in <module>

    write_output(prod)

  File "C:\_DevEnv\vcpkg\buildtrees\glib\x64-windows-dbg\gobject\glib-mkenums", line 97, in write_output

    print(output, file=output_stream)

UnicodeEncodeError: 'gbk' codec can't encode character '\xa9' in position 17: illegal multibyte sequence
````
This seems to be a hardcoded problem, please make sure your VS is using the English language pack and install the build tools. Then clean vcpkg cache and try again.

**Comment by YunchengLiu** [comment]
@FrankXie05 Thanks,  Visual Studio 2022 in my PC has already been installed with English language pack, and i tried to remove the buildtrees directory and rebuild `glib`, the problem is still existed.

**Comment by FrankXie05** [comment]
@YunchengLiu Can you run the command `.\vcpkg.exe install glib:x64-windows --debug` and give me the console output and error log? :)

**Comment by YunchengLiu** [comment]
I updated the vcpkg to the newest version (2022-10-17), and rebuild `glib`, the log and build output files are belows.

Still seems to be the same Unicode error.

I reconfirmed the installation information of my Visual Studio 2022 and confirmed that the English language pack has been installed.

Thank you very much for your help!

[glib_x64_build_log.txt](https://github.com/microsoft/vcpkg/files/9807513/glib_x64_build_log.txt)
[glib_package-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/9807514/glib_package-x64-windows-dbg-err.log)
[glib_package-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/9807515/glib_package-x64-windows-rel-out.log)

**Comment by FrankXie05** [comment]
This is a problem of python hardcoding and `print` restricted encoding format.
You can modify the encoding format of your native python to `utf-8` [here](https://stackoverflow.com/questions/5419/python-unicode-and-the-windows-console)

**Comment by YunchengLiu** [comment]
Thanks. I see in the log that it seems to be related to python bound with vcpkg. Do I need to change the settings of my native python environment? My native python is configured through `conda` and is not added to the global environment variable.

I followed [this link's guide](https://stackoverflow.com/questions/5419/python-unicode-and-the-windows-console), and installed `pip` for vcpkg's python(from [here](https://stackoverflow.com/questions/42666121/pip-with-embedded-python) ), then install the `win-unicode-console` package, rebuild `glib`, there still the same error.

**Comment by FrankXie05** [comment]
@YunchengLiu You need to add your python to environment variables.

**Comment by YunchengLiu** [comment]
I am sorry for that, i added my python to environment variables, and set as the guide said, same error still occurred.

**Comment by YunchengLiu** [comment]
@FrankXie05 Hi, 
The problem was solved, I didn't do anything, after about 2 weeks (date: 2022-11-01 14:30), I synced to the latest upstream and then successfully built glib.

I'm guessing it's because the last couple of updates have updated the embedded python interpreter, fixed some issues I'm unaware of, and either directly or indirectly fixed the aforementioned Unicode decoding bug.

As for my local Python environment, I guess it doesn't matter, because it seems that the python that comes with vcpkg is called first when building. The reason why I say this is because I uninstalled my local python environment (I rarely use it), in The problem still occurs before syncing with vcpkg upstream today.

But anyway, even though I'm still confused and can't figure out why, I'm very grateful for the help of the maintainers before this, this issue can be closed!
